Build of hpctoolkit fails on Fedora 33

Trying a build of hpctoolkit from fresh clone of spack fails on Fedora 33.  No special customizations done.

### Steps to reproduce the issue

#!/bin/bash
git clone https://github.com/spack/spack.git
source spack/share/spack/setup-env.sh
spack install -j 4 hpctoolkit +mpi ^openmpi

### Error Message

==> Installing hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty
==> No binary for hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty found: installing from source
==> hpctoolkit: Executing phase: 'autoreconf'
==> hpctoolkit: Executing phase: 'configure'
==> hpctoolkit: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

1 error found in build log:
     2396    make[5]: *** Waiting for unfinished jobs....
     2397    
     2398    *** Warning: Linking the shared library libhpcrun.la against the
     2399    *** static library /home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/mbedtls-2.16.9-m4yi2faqh7kj3er6synhxtzo25evhjfg/lib/libmbedcrypto.a is not portabl
             e!
     2400    libtool: link: /home/wdn/Packages/spack/spack/lib/spack/env/gcc/gcc -shared  -fPIC -DPIC  utilities/.libs/libhpcrun_la-first_func.o .libs/libhpcrun_la-main.o .libs/libhpcrun_la
             -disabled.o .libs/libhpcrun_la-closure-registry.o .libs/libhpcrun_la-cct_insert_backtrace.o .libs/libhpcrun_la-cct_backtrace_finalize.o .libs/libhpcrun_la-env.o .libs/libhpcrun
             _la-epoch.o .libs/libhpcrun_la-files.o .libs/libhpcrun_la-handling_sample.o .libs/libhpcrun_la-hpcrun-initializers.o .libs/libhpcrun_la-hpcrun_options.o .libs/libhpcrun_la-hpcr
             un_stats.o .libs/libhpcrun_la-loadmap.o .libs/libhpcrun_la-metrics.o .libs/libhpcrun_la-name.o .libs/libhpcrun_la-rank.o .libs/libhpcrun_la-sample_event.o .libs/libhpcrun_la-sa
             mple_prob.o .libs/libhpcrun_la-sample_sources_all.o sample-sources/blame-shift/.libs/libhpcrun_la-blame-shift.o sample-sources/blame-shift/.libs/libhpcrun_la-blame-map.o sample
             -sources/blame-shift/.libs/libhpcrun_la-directed.o sample-sources/blame-shift/.libs/libhpcrun_la-undirected.o sample-sources/.libs/libhpcrun_la-omp-mutex.o sample-sources/.libs
             /libhpcrun_la-omp-idle.o sample-sources/.libs/libhpcrun_la-common.o sample-sources/.libs/libhpcrun_la-display.o sample-sources/.libs/libhpcrun_la-ga.o sample-sources/.libs/libh
             pcrun_la-io.o sample-sources/.libs/libhpcrun_la-itimer.o sample-sources/.libs/libhpcrun_la-idle.o sample-sources/.libs/libhpcrun_la-memleak.o sample-sources/.libs/libhpcrun_la-
             pthread-blame.o sample-sources/.libs/libhpcrun_la-none.o sample-sources/.libs/libhpcrun_la-retcnt.o sample-sources/.libs/libhpcrun_la-sync.o .libs/libhpcrun_la-sample_sources_r
             egistered.o sample-sources/.libs/libhpcrun_la-sample-filters.o .libs/libhpcrun_la-segv_handler.o .libs/libhpcrun_la-start-stop.o .libs/libhpcrun_la-term_handler.o .libs/libhpcr
             un_la-thread_data.o .libs/libhpcrun_la-thread_use.o .libs/libhpcrun_la-thread_finalize.o .libs/libhpcrun_la-control-knob.o .libs/libhpcrun_la-device-finalizers.o .libs/libhpcru
             n_la-device-initializers.o .libs/libhpcrun_la-module-ignore-map.o .libs/libhpcrun_la-threadmgr.o .libs/libhpcrun_la-trace.o .libs/libhpcrun_la-weak.o .libs/libhpcrun_la-write_d
             ata.o cct/.libs/libhpcrun_la-cct_bundle.o cct/.libs/libhpcrun_la-cct_ctxt.o cct/.libs/libhpcrun_la-cct.o cct/.libs/libhpcrun_la-cct-node-vector.o .libs/libhpcrun_la-cct2metrics
             .o lush/.libs/libhpcrun_la-lush-backtrace.o lush/.libs/libhpcrun_la-lush.o lush/.libs/libhpcrun_la-lush-pthread.o lush/.libs/libhpcrun_la-lush-support-rt.o lush/.libs/libhpcrun
             _la-lushi-cb.o fnbounds/.libs/libhpcrun_la-fnbounds_common.o memory/.libs/libhpcrun_la-mem.o memory/.libs/libhpcrun_la-mmap.o messages/.libs/libhpcrun_la-debug-flag.o messages/
             .libs/libhpcrun_la-messages-sync.o messages/.libs/libhpcrun_la-messages-async.o messages/.libs/libhpcrun_la-fmt.o .libs/libhpcrun_la-hpcrun-placeholders.o gpu/.libs/libhpcrun_l
             a-gpu-activity.o gpu/.libs/libhpcrun_la-gpu-activity-channel.o gpu/.libs/libhpcrun_la-gpu-activity-process.o gpu/.libs/libhpcrun_la-gpu-application-thread-api.o gpu/.libs/libhp
             crun_la-gpu-channel-item-allocator.o gpu/.libs/libhpcrun_la-gpu-context-id-map.o gpu/.libs/libhpcrun_la-gpu-correlation.o gpu/.libs/libhpcrun_la-gpu-correlation-channel.o gpu/.
             libs/libhpcrun_la-gpu-correlation-channel-set.o gpu/.libs/libhpcrun_la-gpu-correlation-id.o gpu/.libs/libhpcrun_la-gpu-correlation-id-map.o gpu/.libs/libhpcrun_la-gpu-event-id-
             map.o gpu/.libs/libhpcrun_la-gpu-function-id-map.o gpu/.libs/libhpcrun_la-gpu-host-correlation-map.o gpu/.libs/libhpcrun_la-gpu-metrics.o gpu/.libs/libhpcrun_la-gpu-monitoring.
             o gpu/.libs/libhpcrun_la-gpu-monitoring-thread-api.o gpu/.libs/libhpcrun_la-gpu-op-placeholders.o gpu/.libs/libhpcrun_la-gpu-splay-allocator.o gpu/.libs/libhpcrun_la-gpu-stream
             -id-map.o gpu/.libs/libhpcrun_la-gpu-trace.o gpu/.libs/libhpcrun_la-gpu-trace-channel.o gpu/.libs/libhpcrun_la-gpu-trace-item.o ompt/.libs/libhpcrun_la-ompt-callstack.o ompt/.l
             ibs/libhpcrun_la-ompt-defer.o ompt/.libs/libhpcrun_la-ompt-device.o ompt/.libs/libhpcrun_la-ompt-defer-write.o ompt/.libs/libhpcrun_la-ompt-interface.o ompt/.libs/libhpcrun_la-
             ompt-queues.o ompt/.libs/libhpcrun_la-ompt-region.o ompt/.libs/libhpcrun_la-ompt-region-debug.o ompt/.libs/libhpcrun_la-ompt-placeholders.o ompt/.libs/libhpcrun_la-ompt-device-
             map.o ompt/.libs/libhpcrun_la-ompt-task.o ompt/.libs/libhpcrun_la-ompt-thread.o syscalls/.libs/libhpcrun_la-poll.o syscalls/.libs/libhpcrun_la-ppoll.o syscalls/.libs/libhpcrun_
             la-select.o utilities/.libs/libhpcrun_la-executable-path.o utilities/.libs/libhpcrun_la-hpcrun-nanotime.o utilities/.libs/libhpcrun_la-ip-normalized.o utilities/.libs/libhpcrun
             _la-line_wrapping.o utilities/.libs/libhpcrun_la-timer.o utilities/.libs/libhpcrun_la-tokenize.o utilities/.libs/libhpcrun_la-unlink.o trampoline/common/.libs/libhpcrun_la-tram
             poline_lazy.o sample-sources/perf/.libs/libhpcrun_la-event_custom.o sample-sources/perf/.libs/libhpcrun_la-linux_perf.o sample-sources/perf/.libs/libhpcrun_la-perf_event_open.o
              sample-sources/perf/.libs/libhpcrun_la-perf-util.o sample-sources/perf/.libs/libhpcrun_la-perf_mmap.o sample-sources/perf/.libs/libhpcrun_la-perf_skid.o sample-sources/perf/.l
             ibs/libhpcrun_la-perfmon-util.o sample-sources/perf/.libs/libhpcrun_la-kernel_blocking.o fnbounds/.libs/libhpcrun_la-fnbounds_client.o fnbounds/.libs/libhpcrun_la-fnbounds_dyna
             mic.o monitor-exts/.libs/libhpcrun_la-openmp.o .libs/libhpcrun_la-hpcrun_dlfns.o .libs/libhpcrun_la-custom-init-dynamic.o os/linux/.libs/libhpcrun_la-dylib.o trampoline/x86-fam
             ily/.libs/libhpcrun_la-x86-tramp.o utilities/arch/x86-family/.libs/libhpcrun_la-x86-context-pc.o sample-sources/.libs/libhpcrun_la-papi-c.o sample-sources/.libs/libhpcrun_la-pa
             pi-c-extended-info.o unwind/common/.libs/libhpcrun_la-backtrace.o unwind/common/.libs/libhpcrun_la-unw-throw.o unwind/common/.libs/libhpcrun_la-binarytree_uwi.o unwind/common/.
             libs/libhpcrun_la-interval_t.o unwind/common/.libs/libhpcrun_la-libunw_intervals.o unwind/common/.libs/libhpcrun_la-stack_troll.o unwind/common/.libs/libhpcrun_la-uw_hash.o unw
             ind/common/.libs/libhpcrun_la-uw_recipe_map.o unwind/x86-family/.libs/libhpcrun_la-x86-all.o unwind/x86-family/.libs/libhpcrun_la-amd-xop.o unwind/x86-family/.libs/libhpcrun_la
             -x86-cold-path.o unwind/x86-family/.libs/libhpcrun_la-x86-validate-retn-addr.o unwind/x86-family/.libs/libhpcrun_la-x86-unwind-interval.o unwind/x86-family/.libs/libhpcrun_la-x
             86-unwind-interval-fixup.o unwind/x86-family/.libs/libhpcrun_la-x86-unwind.o unwind/x86-family/.libs/libhpcrun_la-x86-unwind-support.o unwind/x86-family/manual-intervals/.libs/
             libhpcrun_la-x86-gcc-adjust.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-gcc-main64.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-linux-dlresolver.
             o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-intel11-f90main.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-intel-align32.o unwind/x86-family/manual
             -intervals/.libs/libhpcrun_la-x86-intel-align64.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-intel-composer2013-mic.o unwind/x86-family/manual-intervals/.libs/li
             bhpcrun_la-x86-32bit-main.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-32bit-icc-variant.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-fail-interva
             ls.o unwind/x86-family/manual-intervals/.libs/libhpcrun_la-x86-pgi-mp_pexit.o utilities/.libs/libhpcrun_la-last_func.o  -Wl,--whole-archive ../../../src/lib/prof-lean/.libs/lib
             HPCprof-lean.a ../../../src/lib/support-lean/.libs/libHPCsupport-lean.a -Wl,--no-whole-archive  -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/lib
             monitor-2020.10.15-jnmmwudcdapwrhrkdcdz2ggk6txe4b7d/lib -lmonitor -lpthread -lrt -lelf -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/elfutils-0.1
             82-lfnivshxzstkmwfqrd54mmhwa5p5fmer/lib -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/libunwind-1.5.0-som5mkz3qiil66uqvxuhze36ldefbx4s/lib -lunwi
             nd -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/xz-5.2.5-t67ym52gmkgsjx3izsux633b2u3ke2qs/lib -llzma -L/home/wdn/Packages/spack/spack/opt/spack/
             linux-fedora33-haswell/gcc-10.2.1/papi-6.0.0.1-6lwlusuf7g22czl63sthpj5ogsu7v35p/lib -lpfm /home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/mbedtls-2.1
             6.9-m4yi2faqh7kj3er6synhxtzo25evhjfg/lib/libmbedcrypto.a -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/intel-xed-2019.03.01-4q665jwdggy3e544cgncw
             nevqiyytdxm/lib -lpapi -L/home/wdn/Packages/spack/spack/opt/spack/linux-fedora33-haswell/gcc-10.2.1/gotcha-1.0.3-rv644sw5ptqdyhy4nywnhpsbxenjbejg/lib64 -lgotcha -lxed  -g -O3 -
             Wl,-Bsymbolic   -Wl,-soname -Wl,libhpcrun.so.0 -o .libs/libhpcrun.so.0.0.0
     2401    /usr/bin/ld: gpu/.libs/libhpcrun_la-gpu-metrics.o:/tmp/wdn/spack-stage/spack-stage-hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty/spack-src/src/tool/hpcrun/gpu/gpu-metr
             ics.h:66: multiple definition of `gpu_inst_stall_all_t'; gpu/.libs/libhpcrun_la-gpu-application-thread-api.o:/tmp/wdn/spack-stage/spack-stage-hpctoolkit-2020.08.03-dfrulxfvokpy
             et3afosqm3su3rpqkcty/spack-src/src/tool/hpcrun/gpu/gpu-metrics.h:66: first defined here
  >> 2402    collect2: error: ld returned 1 exit status
     2403    make[5]: *** [Makefile:2771: libhpcrun.la] Error 1
     2404    make[5]: Leaving directory '/tmp/wdn/spack-stage/spack-stage-hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty/spack-src/src/tool/hpcrun'
     2405    make[4]: *** [Makefile:7984: all-recursive] Error 1
     2406    make[4]: Leaving directory '/tmp/wdn/spack-stage/spack-stage-hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty/spack-src/src/tool/hpcrun'
     2407    make[3]: *** [Makefile:2043: all] Error 2
     2408    make[3]: Leaving directory '/tmp/wdn/spack-stage/spack-stage-hpctoolkit-2020.08.03-dfrulxfvokpyet3afosqm3su3rpqkcty/spack-src/src/tool/hpcrun'

### Information on your system

% spack debug report
* **Spack:** 0.16.0-444-22a45e010
* **Python:** 3.9.1
* **Platform:** linux-fedora33-haswell
* **Concretizer:** original

No customizations being used.

### Additional information

- [ x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [ ] I have searched the issues of this repo and believe this is not a duplicate
- [ ] I have run the failing commands in debug mode and reported the output

<!-- We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively!

If you want to ask a question about the tool (how to use it, what it can currently do, etc.), try the `#general` channel on our Slack first. We have a welcoming community and chances are you'll get your reply faster and without opening an issue.

Other than that, thanks for taking the time to contribute to Spack! -->

--- a/var/spack/repos/builtin/packages/hpctoolkit/package.py
+++ b/var/spack/repos/builtin/packages/hpctoolkit/package.py
@@ -113,6 +113,13 @@ class Hpctoolkit(AutotoolsPackage):
     conflicts('^binutils@2.35:2.35.1',
               msg='avoid binutils 2.35 and 2.35.1 (spews errors)')
 
+    # Fixes multiple definition error with GCC 10.
+    # https://github.com/HPCToolkit/hpctoolkit/issues/309
+    patch('https://github.com/blue42u/hpctoolkit/commit/'
+          'b3f6f9e4846d9256cf0d841465ff89d78c6bf422.patch',
+          when='%gcc@10',
+          sha256='f8507c3ce9672c70c2db9f9deb5766c8120ea06e20866d0f553a17866e810b91')
+
     flag_handler = AutotoolsPackage.build_system_flags
 
     def configure_args(self):

